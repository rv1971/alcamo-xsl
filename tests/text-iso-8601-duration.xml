<?xml version="1.0" encoding="UTF-8"?>

<?xml-stylesheet href="#xslt" type="text/xml"?>

<a:doc xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#">
  <xsl:stylesheet
      xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
      xml:id="xslt"
      version="1.0">
    <xsl:import href="../xsl/text.xsl"/>

    <xsl:template match="/">
      <a:result>
        <xsl:apply-templates select="/a:doc/a:data"/>
      </a:result>
    </xsl:template>

    <xsl:template match="a:item">
      <a:item>
        <xsl:value-of select="."/>
        <xsl:value-of select="' '"/>
        <xsl:apply-templates select="." mode="a:iso-8601-duration"/>
      </a:item>
    </xsl:template>
  </xsl:stylesheet>

  <a:data>
    <a:item>P1D</a:item>
    <a:item>P1W2D</a:item>
    <a:item>P1M2D</a:item>
    <a:item>P1Y2M3D</a:item>
    <a:item>PT2H</a:item>
    <a:item>PT2H3M</a:item>
    <a:item>PT2H3M4S</a:item>
    <a:item>PT2H3M4.567890S</a:item>
    <a:item>P7Y5DT4M1.3S</a:item>
    <a:item>P3WT1S</a:item>
  </a:data>
</a:doc>
