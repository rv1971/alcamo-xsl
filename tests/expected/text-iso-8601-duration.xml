<?xml version="1.0"?>
<a:result xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#">
  <a:item>P1D 1&#xA0;day</a:item>
  <a:item>P1W2D 1&#xA0;week 2&#xA0;days</a:item>
  <a:item>P1M2D 1&#xA0;month 2&#xA0;days</a:item>
  <a:item>P1Y2M3D 1&#xA0;year 2&#xA0;months 3&#xA0;days</a:item>
  <a:item>PT2H 2&#xA0;hours</a:item>
  <a:item>PT2H3M 2&#xA0;hours 3&#xA0;minutes</a:item>
  <a:item>PT2H3M4S 2&#xA0;hours 3&#xA0;minutes 4&#xA0;seconds</a:item>
  <a:item>PT2H3M4.567890S 2&#xA0;hours 3&#xA0;minutes 4.567890&#xA0;seconds</a:item>
  <a:item>P7Y5DT4M1.3S 7&#xA0;years 5&#xA0;days 4&#xA0;minutes 1.3&#xA0;seconds</a:item>
  <a:item>P3WT1S 3&#xA0;weeks 1&#xA0;second</a:item>
</a:result>
